### PR TITLE
Persist PivotChart state

### LINKED ISSUE #379: Persist state of PivotChart (pivottable) [doing]
PivotChart does not persist it's configuration (selected graph type, dimensions, etc). it would be very nice to persist that, so that on notebook page reload we'd see the same view? it's rather annoying now...

@andypetrella

---

 http://nicolas.kruchten.com/pivottable/examples/onrefresh.html here's an example on how to fetch state from pivottable. it would be good to save/persist this on every notebook save/checkpoint.

and then to restore the state looks like one needs to pass it here: https://github.com/nicolaskruchten/pivottable/wiki/Parameters#pivotuiinput-options-overwrite-locale

>  sounds good, actually the js+json thingy are stored in the output of the cell which for now is only updated from the web socket response from the original code. We shall add a JS function/ality that can also update this content "live". Need to check this out deeper to come up with a proposal though

### PR DESCRIPTION
**Pull request by vidma**
@andypetrella 

seem to work OK both then saving/restoring state, and when state was not available.

bare in mind I never used `coffeescript`, so suggestions for improvements are more than welcome.

fixes #379

### CODE REVIEW

File: app/assets/javascripts/notebook/magic/pivotChart.coffee [hunk 1]
```diff
@@ -20,9 +20,38 @@ define([
...
                                 )
+
+      get_cell = () ->
+        # get cell object, like done in get_cell_elements in notebook.js
+        cell_dom_element = $(container).parents(".cell").not('.cell .cell')[0]
+        $(cell_dom_element).data("cell")
+
+      get_saved_pivot_state = () ->
+        saved_state = get_cell().metadata.presentation?.pivot_chart_state
+        JSON.parse(saved_state || "{}") || {}
```
Comment (vidma) [inline app/assets/javascripts/notebook/magic/pivotChart.coffee]: if for some weird reason saved state was non empty & not JSON parseable, this raise exception

**Comment by ljank** [comment]
Lack of tests makes me shiver :(

**Comment by andypetrella** [comment]
@ljank agreed! Do you have any XP in this kind of JS tests? I have very little, and it would take me so long to put something in case which will be probably too naive :-(

**Comment by ljank** [comment]
I have some noobish experience with JS tests. Not sure if it's the best option, but I was very productive and happy using `mocha` accompanied by `assert` and `sinon` for spies/mocks. If you're OK with such combo, I could try and set it up.

**Comment by vidma** [comment]
I'd say :ship:

**Pull request merged**